File: 4th Year/2nd Semester/Perfil 1 - Criptografia e Segurança da Informação/Engenharia de Segurança/Trabalhos Práticos/Antigos/Grupo2/TP7/ReadTemps_fix.java

import java.io.*;  // File
import java.util.*;  // Scanner

public class ReadTemps
{
    public static void main (String[] args) throws FileNotFoundException
    {
        Scanner inFile = new Scanner(new File("/tmp/temps.txt"));  
        double[] temps = new double[10];
        int numTemps = 0;

        while (inFile.hasNextDouble())
        {
            if(numTemps < 10){
                temps[numTemps] = inFile.nextDouble();
                numTemps++;
            }else break;
        }
        System.out.println("Foram lidas " + numTemps + " temperaturas.");

    }
}

File: 4th Year/2nd Semester/Perfil 1 - Criptografia e Segurança da Informação/Engenharia de Segurança/Trabalhos Práticos/Antigos/Grupo2/TP7/ReadTemps_fix.py
temps = [None]*10
numTemps = 0
infile = open("/tmp/temps.txt","r")           
for line in infile.readlines():         
    for i in line.split():            
        if numTemps < 10:
            temps[numTemps] = float(i)   
            numTemps += 1
        else:
            break
print('Foram lidas ' + str(numTemps) + ' temperaturas.')
